test/parser_test.cpp

```diff
@@ -0,0 +1,20 @@
+#include "yaml-cpp/parser.h"
+#include "mock_event_handler.h"
+#include "gtest/gtest.h"
+
+using YAML::Parser;
+using YAML::MockEventHandler;
+
+TEST(ParserTest, Empty) {
+    Parser parser;
+
+    EXPECT_FALSE(parser);
+
+    MockEventHandler handler;
+    EXPECT_FALSE(parser.HandleNextDocument(handler));
+    
+    std::ostringstream oss;
+    parser.PrintTokens(oss);
```

Comment:
Maybe instead of checking the token list, verify that there are no interactions with the MockEventHandler?

Reply:
@jbeder Yes, good proposal! I will do it. I have some doubts about doing it _instead_. I would add it as an addition to current checks. What do you think?

Reply:
The reason I'd suggest *instead* is that Parser::PrintTokens is a debugging function (it's actually unused, and probably should just be deleted...), so it's not good to rely on its output in a test.

Reply:
@jbeder I have done it